src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoBaseSink.java

```diff
@@ -0,0 +1,211 @@
+     * Available data models when persisting data at Mongo.
+     */
+    public enum DataModel { COLLECTIONPERSERVICEPATH, COLLECTIONPERENTITY, COLLECTIONPERATTRIBUTE }
+    
+    protected static final CygnusLogger LOGGER = new CygnusLogger(OrionMongoBaseSink.class);
+    protected String mongoHosts;
+    protected String mongoUsername;
+    protected String mongoPassword;
+    protected DataModel dataModel;
+    protected String dbPrefix;
+    protected String collectionPrefix;
+    protected MongoBackend backend;
+    
+    /**
+     * Gets the mongo URI. It is protected since it is used by the tests.
+     * @return
+     */
+    protected String getURI() {
+        return mongoHosts;
+    } // getURI
+    
+    /**
+     * Gets the mongo username. It is protected since it is used by the tests.
+     * @return
+     */
+    protected String getUsername() {
+        return mongoUsername;
+    } // getUsername
+    
+    /**
+     * Gets the mongo password. It is protected since it is used by the tests.
+     * @return
+     */
+    protected String getPassword() {
+        return mongoPassword;
+    } // getPassword
+    
+    /**
+     * Gets the mongo data model. It is protected since it is used by the tests.
+     * @return
+     */
+    protected DataModel getDataModel() {
+        return dataModel;
+    } // getDataModel
+    
+    /**
+     * Gets the database prefix. It is protected since it is used by the tests.
+     * @return
+     */
+    protected String getDbPrefix() {
+        return dbPrefix;
+    } // getDBPrefix
+    
+    /**
+     * Gets the collection prefix. It is protected since it is used by the tests.
+     * @return
+     */
+    protected String getCollectionPrefix() {
+        return collectionPrefix;
+    } // getCollectionPrefix
+    
+    /**
+     * Sets the backend. It is protected since it is used by the tests.
+     * @param backend
+     */
+    protected void setBackend(MongoBackend backend) {
+        this.backend = backend;
+    } // setBackend
+    
+    /**
+     * Gets the backend. It is protected since it is used by the tests.
+     * @return
+     */
+    protected MongoBackend getBackend() {
+        return backend;
+    } // getBackend
+    
+    @Override
+    public void configure(Context context) {
+        mongoHosts = context.getString("mongo_hosts", "localhost:27017");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (mongo_hosts=" + mongoHosts + ")");
+        mongoUsername = context.getString("mongo_username", "");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (mongo_username=" + mongoUsername + ")");
+        // FIXME: mongoPassword should be read as a SHA1 and decoded here
+        mongoPassword = context.getString("mongo_password", "");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (mongo_password=" + mongoPassword + ")");
+        dataModel = getDataModel(context.getString("data_model", "collection-per-attribute"));
```

Comment:
In the STH component, the default data model is collection-per-entity. We should align ourselves here :)